WaterLog_Backend/Controllers/SegmentLeaksController.cs

```diff
@@ -61,24 +61,28 @@ public async Task<ActionResult<string>> GetLitres(int id)
 
         //Resolve Leakage
         [HttpPut("resolve/{id}")]
-        public async Task<IActionResult> Resolve(int id)
+        public async Task<ActionResult<IEnumerable<SegmentLeaksEntry>>> Resolve(int id)
         {
-            var leaks = await _db.SegmentLeaks.FindAsync(id);
+            var leaks = await _db.SegmentLeaks.Where(s => s.SegmentsId == id).FirstOrDefaultAsync();
             if (leaks == null)
             {
                 return NotFound();
             }
-            leaks.ResolvedStatus = "resolved";
-            _db.SegmentLeaks.Update(leaks);
+            if (leaks.ResolvedStatus == "unresolved")
+            {
+                leaks.ResolvedStatus = "resolved";
```

Comment:
So I want to know where else are you actually giving ResolvedStatus a value of String? because if its just this file then I can resolve this comment but if its more than 1 file then I'd like you to put the statuses in an Enum.

Reply:
On procedures, but that's just for our data stream

Reply:
You should actually add them to enum for learning purpose

Reply:
Working on it

Reply:
Done, made the resolveStatus an Enum

Reply:
Good work, thanks